New board nucleo g431kb (#78091)
zephyrproject-rtos/zephyr: boards/st/nucleo_g431kb/doc/index.rst

.. _nucleo_g431kb_board:

ST Nucleo G431KB
################

Overview
********

The Nucleo G431kB board features an ARM Cortex-M4 based STM32G431kB MCU
with a wide range of connectivity support and configurations.
Here are some highlights of the Nucleo G431kB board:

- STM32 microcontroller in LQFP32 package
- Arduino nano V3 connectivity
- On-board ST-LINK/V3E debugger/programmer
- Flexible board power supply:

  - USB VBUS or external source(3.3V, 5V, 7 - 12V)
  - Power management access point

- Three LEDs: USB communication (LD1), power LED (LD3), user LED (LD2)
- One push-button for RESET

.. image:: img/nucleo_g431kb.jpg
  :align: center
  :alt: Nucleo G431kB

More information about the board can be found at the `Nucleo G431KB website`_.

- Development support: serial wire debug (SWD), JTAG, Embedded Trace Macrocell*

More information about STM32G431RB can be found here:

- `STM32G431KB on www.st.com`_
- `STM32G4 reference manual`_

Supported Features
==================

> The Zephyr nucleo_g431kb board configuration supports the following hardware features:

Reviewer: use \`\` around board name and change to board target

Author: What do you mean with board target?

Reviewer: board configuration does not exist, see https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#board-terminology

Author: do you mean like ...  `nucleo_g431kb` board target.  ... or do you mean using a specific reST role or a reference?

Reviewer: ```suggestion
The Zephyr ``nucleo_g431kb`` board target supports the following hardware features:
```

:zephyr_file:`boards/st/nucleo_g431kb/nucleo_g431kb_defconfig`

Connections and IOs
===================

Nucleo G431KB Board has 6 GPIO controllers. These controllers are responsible for pin muxing,
input/output, pull-up, etc.

For more details please refer to `STM32G4 Nucleo-32 board User Manual`_.

Default Zephyr Peripheral Mapping:
----------------------------------

.. rst-class:: rst-columns

- LPUART_1_TX : PA2
- LPUART_1_RX : PA3
- LD2 : PB8

System Clock
------------

The Nucleo G431KB System Clock could be driven by internal or external oscillator,
as well as main PLL clock. By default the external oscillator is not connected to the board. Therefore only the internal
High Speed oscillator is supported. By default System clock is driven by PLL clock at 170MHz,
the PLL is driven by the 16MHz high speed internal oscillator.

Serial Port
-----------

Nucleo G431KB board has 1 U(S)ARTs and one LPUART. The Zephyr console output is assigned to LPUART1.
Default settings are 115200 8N1.

Please note that LPUART1 baudrate is limited to 9600 if the MCU is clocked by LSE (32.768 kHz) in
low power mode.

Programming and Debugging
*************************

> Applications for the ``nucleo_g431kb`` board configuration can be built and

Reviewer: board target

.. _nucleo_g431kb_board:

ST Nucleo G431KB
################

Overview
********

> The Nucleo G431kB board features an ARM Cortex-M4 based STM32G431kB MCU

Reviewer 2: Maybe a spell checker played tricks on you? Just noticed this sorry

```suggestion
The Nucleo G431KB board features an ARM Cortex-M4 based STM32G431KB MCU
```

Author: The port started with a copy from the Nucleo G431RB board and some find and replace actions

with a wide range of connectivity support and configurations.

> Here are some highlights of the Nucleo G431kB board:

Reviewer 3: ```suggestion
Here are some highlights of the Nucleo G431KB board:
```

- STM32 microcontroller in LQFP32 package

> - Arduino nano V3 connectivity

Reviewer 2: ```suggestion
- Arduino Nano V3 connectivity
```

- On-board ST-LINK/V3E debugger/programmer
- Flexible board power supply:

>   - USB VBUS or external source(3.3V, 5V, 7 - 12V)

Reviewer 2: ```suggestion
  - USB VBUS or external source(3.3 V, 5 V, 7-12 V)
```

  - Power management access point

- Three LEDs: USB communication (LD1), power LED (LD3), user LED (LD2)
- One push-button for RESET

.. image:: img/nucleo_g431kb.webp
  :align: center
  :alt: Nucleo G431kB

More information about the board can be found at the `Nucleo G431KB website`_.

> - Development support: serial wire debug (SWD), JTAG, Embedded Trace Macrocell*

Reviewer 2: Assuming this is a copy-paste from somewhere, the asterisk probably doesn't make sense here?

```suggestion
- Development support: serial wire debug (SWD), JTAG, Embedded Trace Macrocell.
```

> More information about STM32G431RB can be found here:

Reviewer 2: ```suggestion
More information about STM32G431KB can be found here:
```

+-----------+------------+-------------------------------------+
| PWM       | on-chip    | pwm                                 |
+-----------+------------+-------------------------------------+
| I2C       | on-chip    | i2c                                 |
+-----------+------------+-------------------------------------+

Other hardware features are not yet supported on this Zephyr port.

The default configuration can be found in the defconfig file:
:zephyr_file:`boards/st/nucleo_g431kb/nucleo_g431kb_defconfig`

Connections and IOs
===================

Nucleo G431KB Board has 6 GPIO controllers. These controllers are responsible for pin muxing,
input/output, pull-up, etc.

For more details please refer to `STM32G4 Nucleo-32 board User Manual`_.

Default Zephyr Peripheral Mapping:
----------------------------------

.. rst-class:: rst-columns

- LPUART_1_TX : PA2
- LPUART_1_RX : PA3
- LD2 : PB8
- PWM_4_CH_3 : PB8
- I2C_2_SCL : PA9
- I2C_2_SDA : PA8

System Clock
------------

The Nucleo G431KB System Clock could be driven by internal or external oscillator,
as well as main PLL clock. By default the external oscillator is not connected to the board. Therefore only the internal

> High Speed oscillator is supported. By default System clock is driven by PLL clock at 170MHz,
> the PLL is driven by the 16MHz high speed internal oscillator.

Reviewer 2: ```suggestion
High Speed oscillator is supported. By default System clock is driven by PLL clock at 170 MHz,
the PLL is driven by the 16 MHz high speed internal oscillator.
```

Serial Port
-----------

Nucleo G431KB board has 1 U(S)ARTs and one LPUART. The Zephyr console output is assigned to LPUART1.
Default settings are 115200 8N1.

Please note that LPUART1 baudrate is limited to 9600 if the MCU is clocked by LSE (32.768 kHz) in
low power mode.

Programming and Debugging
*************************

Nucleo G431KB Board includes an ST-Link/V3 embedded debug tool interface.

Applications for the ``nucleo_g431kb`` board target can be built and
flashed in the usual way (see :ref:`build_an_application` and
:ref:`application_run` for more details).

Flashing
========

The board is configured to be flashed using west `STM32CubeProgrammer`_ runner,
so its :ref:`installation <stm32cubeprog-flash-host-tools>` is required.

Alternatively, OpenOCD, or pyOCD can also be used to flash the board using
the ``--runner`` (or ``-r``) option:

.. code-block:: console

   $ west flash --runner openocd
   $ west flash --runner pyocd

> To enable support of the stm32g342kb SoC in pyocd the pyocd pack for this soc has to be installed first:

Reviewer 3: ```suggestion
To enable support for STM32G431KB SoC in pyOCD, its pack has to be installed first:
```
Have you verified that this pack is actually available?

Author: Yes, i use the configuration since you suggested to use the pyocd runner as default. I just ckecked it again, it gets pulled in via Keil.STM32G4xx_DFP.1.6.0

.. code-block:: console

   $ pyocd pack --update

>    $ pyocd pack --install stm32g431rb

Reviewer 3: ```suggestion
   $ pyocd pack --install stm32g431kb
```

> Flashing an application to Nucleo G431kB
> ----------------------------------------
> 
> Connect the Nucleo G431kB to your host computer using the USB port,

Reviewer 2: ```suggestion
Flashing an application to Nucleo G431KB
----------------------------------------

Connect the Nucleo G431KB to your host computer using the USB port,
```

then run a serial host program to connect with your Nucleo board.

.. code-block:: console

   $ minicom -D /dev/ttyACM0

Now build and flash an application. Here is an example for
:zephyr:code-sample:`hello_world`.

.. zephyr-app-commands::
   :zephyr-app: samples/hello_world
   :board: nucleo_g431kb
   :goals: build flash

You should see the following message on the console:

.. code-block:: console

   $ Hello World! nucleo_g431kb/stm32g431xx

Debugging
=========

You can debug an application in the usual way.  Here is an example for the
:zephyr:code-sample:`hello_world` application.

.. zephyr-app-commands::
   :zephyr-app: samples/hello_world

>    :board: nucleo_g431rb

Reviewer 2: ```suggestion
   :board: nucleo_g431kb
```
